File: server/datasets/java_source-code/Multiple-C1/organic_5.java
/*
 * Graph BFS - Organic Submission #5
 * Object-oriented Graph class exposing addEdge() and bfs() methods.
 */

import java.util.*;

class Graph {
    Map<String, List<String>> adjacency;

    public Graph() {
        this.adjacency = new HashMap<>();
    }

    public void addEdge(String u, String v) {
        this.adjacency.computeIfAbsent(u, k -> new ArrayList<>()).add(v);
        this.adjacency.computeIfAbsent(v, k -> new ArrayList<>()).add(u);
    }

    public List<String> bfs(String start) {
        Set<String> visited = new HashSet<>();
        Queue<String> queue = new ArrayDeque<>();
        List<String> path = new ArrayList<>();

        visited.add(start);
        queue.add(start);

        while (!queue.isEmpty()) {
            String vertex = queue.poll();
            path.add(vertex);

            List<String> neighbors = this.adjacency.getOrDefault(vertex, Collections.emptyList());
            for (String neighbor : neighbors) {
                if (!visited.contains(neighbor)) {
                    visited.add(neighbor);
                    queue.add(neighbor);
                }
            }
        }
        return path;
    }

    public static void main(String[] args) {
        Graph g = new Graph();
        String[][] edges = {
            {"A", "B"}, {"A", "C"}, {"B", "D"}, {"C", "D"}, {"D", "E"}
        };
        for (String[] edge : edges) {
            g.addEdge(edge[0], edge[1]);
        }
        System.out.println(g.bfs("A"));
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_c1/organic_5.py
"""
Graph BFS - Organic Submission #5
Object-oriented Graph class exposing an add_edge() and bfs() method.
"""

from collections import deque


class Graph:
    def __init__(self):
        self.adjacency = {}

    def add_edge(self, u, v):
        self.adjacency.setdefault(u, []).append(v)
        self.adjacency.setdefault(v, []).append(u)

    def bfs(self, start):
        visited = {start}
        queue = deque([start])
        path = []

        while queue:
            vertex = queue.popleft()
            path.append(vertex)

            for neighbor in self.adjacency.get(vertex, []):
                if neighbor not in visited:
                    visited.add(neighbor)
                    queue.append(neighbor)

        return path


if __name__ == "__main__":
    g = Graph()
    for u, v in [("A", "B"), ("A", "C"), ("B", "D"), ("C", "D"), ("D", "E")]:
        g.add_edge(u, v)

    print(g.bfs("A"))
